config/memory_statistics.py

```diff
@@ -0,0 +1,146 @@
+import syslog
+
+import click
+from swsscommon.swsscommon import ConfigDBConnector
+
+# Constants
+MEMORY_STATISTICS_TABLE = "MEMORY_STATISTICS"
+MEMORY_STATISTICS_KEY = "memory_statistics"
+SAMPLING_INTERVAL_MIN = 3
+SAMPLING_INTERVAL_MAX = 15
+RETENTION_PERIOD_MIN = 1
+RETENTION_PERIOD_MAX = 30
+DEFAULT_SAMPLING_INTERVAL = 5  # minutes
+DEFAULT_RETENTION_PERIOD = 15  # days
+
+
+def log_to_syslog(message, level=syslog.LOG_INFO):
+    """Log a message to syslog."""
+    syslog.openlog("memory_statistics", syslog.LOG_PID | syslog.LOG_CONS, syslog.LOG_USER)
+    syslog.syslog(level, message)
+
+
+def get_db_connection():
```

Comment:
Seems like every command we are creating a new connection to config db. Perhaps we can create a class variable and create only one connection and reuse?

Reply:
Thanks for the feedback. I have implemented a Singleton pattern for the database connection, ensuring a single connection is reused across all commands.